src/formatDate.js

```diff
@@ -50,7 +50,74 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let year;
+  let month;
+  let day;
+  let separator = fromFormat[3];
+  const fromArray = date.split(separator);
+  const resultArray = [];
+
+  for (let i = 0; i < 3; i++) {
+    const current = fromArray[i];
+
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = current;
+        break;
+
+      case 'MM':
+        month = current;
+        break;
+
+      case 'YY':
+        year = current;
+        break;
+
+      case 'YYYY':
+        year = current;
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
+    switch (toFormat[i]) {
+      case 'DD':
+        resultArray.push(day);
+        break;
+
+      case 'MM':
+        resultArray.push(month);
+        break;
+
+      case 'YY':
+        if (year.length === 4) {
+          year = year.slice(-2);
+        }
+        resultArray.push(year);
+        break;
+
+      case 'YYYY':
+        if (year.length === 2) {
+          if (+year < 30) {
+            year = '20' + year;
+          } else {
+            year = '19' + year;
+          }
+        }
+        resultArray.push(year);
+        break;
+
```

Comment:
you do the same push on each step, can you push it after switch/case ends?

Reply:
> you do the same push on each step, can you push it after switch/case ends?

I think i can't... i do push in each step for the right consistency of date elements. i iterate toFormat. if on zero index in toFormat it's YY, then case 'YY' will work and will push the year on the right place in result array. if i push after all cases i don't know were have to be year, month and day. i need to add more if conditions or something like that. But i can replace push method in each case with `resultArray[i] = day / month/ year`, but it will be the same logic. Maybe there is another solution for this but now i can't remember anything, i`m sorry. maybe you can suggest something 😸 @illiarerun